website/docs/cdktf/examples-and-guides/refactoring.mdx

```diff
@@ -0,0 +1,255 @@
+---
+page_title: Refactoring - CDK for Terraform
+description: >-
+  As your CDKTF application grows in size and complexity, the need to move resources between stacks becomes more common. This guide will walk you through the process of refactoring your stacks.
+---
+
+# Refactoring
+
+As your CDKTF application grows in size and complexity, the need to move resources between stacks becomes more common. This guide will walk you through the process of refactoring your stacks. In general you can change the names of stacks freely without an effect on the synthesized code. If you change the name of a resource this might lead to a re-creation of the resource, especially if you move the resource from one stack to another.
```

Comment:
It might be good after this first sentence to give one real-world example that would make it a good idea to move resources around. Like "For example, you may need to z, y, z." Can we think of anything like that?

```suggestion
You may need to move resources between stacks as your CDK for Terraform (CDKTF) application grows in size and complexity. This guide demonstrates how to refactor your stacks. In general you can change the names of stacks freely without an effect on the synthesized code. If you change the name of a resource this might lead to a re-creation of the resource, especially if you move the resource from one stack to another.
```